apps/asap-server/src/controllers/events.ts

```diff
@@ -74,35 +87,35 @@ export default class Events implements EventController {
 
     if (groupId) {
       const { findGroupsContent } = await this.client.request<
-        ResponseFetchGroup,
-        { id: string }
-      >(buildGraphQLQueryFetchGroup(), { id: groupId });
+        FetchGroupCalendarQuery,
+        FetchGroupCalendarQueryVariables
+      >(FETCH_GROUP_CALENDAR, {
+        id: groupId,
+      });
 
       if (!findGroupsContent) {
         throw Boom.notFound();
       }
 
-      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-      const calendarIds = findGroupsContent.flatData!.calendars!.map(
-        (calendar) => `'${calendar.id}'`,
+      const calendarIds = (findGroupsContent.flatData.calendars ?? []).map(
+        ({ id }) => `'${id}'`,
       );
 
       filters.push(`data/calendar/iv in [${calendarIds.join(', ')}]`);
     }
 
-    const query = buildGraphQLQueryFetchEvents();
-
-    const { queryEventsContentsWithTotal } = await this.client.request<
-      ResponseFetchEvents,
-      GraphqlFetchOptions
-    >(query, {
+    const result = await this.client.request<
+      FetchEventsQuery,
+      FetchEventsQueryVariables
+    >(FETCH_EVENTS, {
       filter: filters.join(' and '),
       top: take,
       skip,
       order: orderby,
     });
 
-    const { total, items: events } = queryEventsContentsWithTotal;
+    const total = result.queryEventsContentsWithTotal?.total ?? 0;
+    const events = result.queryEventsContentsWithTotal?.items ?? [];
```

Comment:
I had copied what @peterstarling did on research-outputs, the return will be the same but we are logging a warning when either `queryEventsContentsWithTotal` or the `item` are null